### PR TITLE
Nc news6

### PR DESCRIPTION
**Pull request by HarrietHall**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -84,3 +80,49 @@ test("404 : Responds with message -'Not Found' when article id is valid but does
...
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: Responds with an array of article comments for the specified article id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+        comments.forEach((comment) => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js:92]: As we're asserting in a forEach, could we check the length beforehand please?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -84,3 +80,49 @@ test("404 : Responds with message -'Not Found' when article id is valid but does
...
+  });
+  test("200: Responds with an empty array if article_id exists but there are no comments ", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments.msg).toBe("No comments found");
+      });
+  });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:111]: Great testing! 👍  Could we also just add another expect to this 200 test, which shows the comments are an empty array please?

File: db/controllers/app.controller.js [hunk 3]
```diff
@@ -15,18 +19,27 @@ exports.getEndpointData = (req, res, next) => {
...
 };
 
+exports.getArticleComments = (req, res, next) => {
+  const { article_id } = req.params;
+  const promises = [selectArticleById(article_id), selectArticleComments(article_id)];
 
+  Promise.all(promises)
 
+    .then((resolvedPromises) => {
+      const comments = resolvedPromises[1];
```
Comment (johnhutcheon) [inline db/controllers/app.controller.js:39]: Really nice work! 🙌

**Review by johnhutcheon — APPROVED** [review]
Hey Harriet, really nice work! Could you just amend the test as per my comment please, then happy for you to merge 🙂

File: __tests__/app.test.js (hunk 2 above)
Comment (HarrietHall) [reply __tests__/app.test.js:111]: HI, thank you for the feedback.  im not sure how to do this because the response which comes back is an object and i cant find the property would show the empty array. the only way i know is to add a comment poperty onto here: 
   return Promise.resolve({ status: 200, comments: [], msg: "No comments found" });?

Comment (HarrietHall) [reply __tests__/app.test.js:111]: this seemed to solve it .... Am i okay to merge like this? thanks 

  exports.selectArticleComments = (article_id) => {
  const query =
    "SELECT * FROM comments WHERE article_id = $1 ORDER BY comments.created_at DESC";

  return db.query(query, [article_id]).then(({ rows }) => {
    if (!rows.length) {
      return Promise.resolve({ comments: `${rows}`,status: 200, msg: "No comments found" });
    }

    return rows;
  });
};test("200: Responds with an empty array if article_id exists but there are no comments ", () => {
    return request(app)
      .get("/api/articles/2/comments")
      .expect(200)
        .then(({body}) => {
        const { comments } = body;
        expect(comments.comments).toHaveLength(0)
        expect(comments.msg).toBe("No comments found");
      });
  });

**Review by robert112233 — APPROVED** [review]
Great work! Merge
